### PR TITLE
Store plugins translations in *.Plugins.xlf

### PR DESCRIPTION
**Pull request by RussKie**
The source of translations was determined based on assembly locations. However, the check was done against a POSIX path, whereas assembly.Location contains Windows native path.

Also, the check was needlessly performed inside a loop.

Resolves https://github.com/gitextensions/gitextensions/pull/10524#discussion_r1049621450

/cc: @ierof

### CODE REVIEW

File: GitUI/Translation/English.Plugins.xlf
Comment (mstv) [inline GitUI/Translation/English.Plugins.xlf:416]: I guess this goes too far. 🤔 These translations seem to have got lost.

File: GitUI/Translation/English.Plugins.xlf
Comment (mstv) [inline GitUI/Translation/English.Plugins.xlf:13]: Why has this changed?

File: ResourceManager/Xliff/TranslationUtil.cs
Comment (mstv) [inline ResourceManager/Xliff/TranslationUtil.cs:302]: This might cause why Forms of plugins are not translated anymore.

File: GitUI/Translation/English.Plugins.xlf
Comment (gerhardol) [reply GitUI/Translation/English.Plugins.xlf:416]: Not a new problem it seems, the texts are not in English.xlf either...

File: GitUI/Translation/English.Plugins.xlf
Comment (gerhardol) [reply GitUI/Translation/English.Plugins.xlf:13]: #8995 changed this

**Review by gerhardol — APPROVED** [review]
Missing FindLargeFilesForm is an issue, but not new and I suggest we merge this now unless someone want to dig in deeper immediately.

File: ResourceManager/Xliff/TranslationUtil.cs
Comment (RussKie) [reply ResourceManager/Xliff/TranslationUtil.cs:302]: No, this is the fix for https://ci.appveyor.com/project/gitextensions/gitextensions/builds/45774723#L346
```
System.NotSupportedException : The invoked member is not supported in a dynamic assembly.
   at System.Reflection.Emit.InternalAssemblyBuilder.get_Location()
   at ResourceManager.Xliff.TranslationUtil.GetTranslatableTypes() in C:\projects\gitextensions\ResourceManager\Xliff\TranslationUtil.cs:line 309
   at GitUITests.TranslationTest.CreateInstanceOfClass() in C:\projects\gitextensions\UnitTests\GitUI.Tests\TranslationTest.cs:line 37
```

File: GitUI/Translation/English.Plugins.xlf
Comment (RussKie) [reply GitUI/Translation/English.Plugins.xlf:416]: The type is getting collected:
image

...however, it can't be instantiated for the translations to get processed:
image

**Review by gerhardol — APPROVED** [review]
Any part of this for 4.0.x or do we leave those translations out of it?
Or should we make a quick 4.1?

File: GitUI/Translation/English.Plugins.xlf
Comment (mstv) [inline GitUI/Translation/English.Plugins.xlf:1178]: I hate error-hiding!

image

```
  ERROR getting types of d:\Build\gitextensions3_dev\artifacts\Debug\bin\GitExtensions\net6.0-windows\Plugins\JiraCommitHintPlugin\GitExtensions.Plugins.JiraCommitHintPlugin.dll:
  System.Reflection.ReflectionTypeLoadException: Unable to load one or more of the requested types.
  Could not load file or assembly 'Atlassian.Jira, Version=12.4.0.0, Culture=neutral, PublicKeyToken=null'. Das System kann die angegebene Datei nicht finden.
     at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
     at System.Reflection.Assembly.GetTypes()
     at ResourceManager.Xliff.TranslationUtil.GetLoadableTypes(Assembly assembly) in d:\Build\gitextensions3_dev\ResourceManager\Xliff\TranslationUtil.cs:line 368
  System.IO.FileNotFoundException: Could not load file or assembly 'Atlassian.Jira, Version=12.4.0.0, Culture=neutral, PublicKeyToken=null'. Das System kann die angegebene Datei nicht finden.
  File name: 'Atlassian.Jira, Version=12.4.0.0, Culture=neutral, PublicKeyToken=null'
  Processing the following types anyway:
  Microsoft.CodeAnalysis.EmbeddedAttribute
  System.Runtime.CompilerServices.NullableAttribute
  System.Runtime.CompilerServices.NullableContextAttribute
  Microsoft.Validates
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin
  GitExtensions.Plugins.JiraCommitHintPlugin.Properties.Resources
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+JiraTaskDTO
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<<Execute>b__22_0>d
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<>c
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<>c__DisplayClass25_0
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<>c__DisplayClass30_0
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<>c__DisplayClass30_1
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<>c__DisplayClass32_0
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<GetSettings>d__23
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<>c__DisplayClass25_0+<<btnPreviewClick>b__0>d
  GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin+<>c__DisplayClass30_0+<<gitUiCommands_PreCommit>b__0>d
  ERROR instantiating type GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin:
  System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
   ---> System.IO.FileNotFoundException: Could not load file or assembly 'Atlassian.Jira, Version=12.4.0.0, Culture=neutral, PublicKeyToken=null'. Das System kann die angegebene Datei nicht finden.
  File name: 'Atlassian.Jira, Version=12.4.0.0, Culture=neutral, PublicKeyToken=null'
     at GitExtensions.Plugins.JiraCommitHintPlugin.JiraCommitHintPlugin..ctor()
     at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
     --- End of inner exception stack trace ---
     at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean wrapExceptions)
     at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
     at System.Activator.CreateInstance(Type type, Boolean nonPublic)
     at ResourceManager.Xliff.TranslationUtil.CreateInstanceOfClass(Type type) in d:\Build\gitextensions3_dev\ResourceManager\Xliff\TranslationUtil.cs:line 350
     at TranslationApp.TranslationHelpers.LoadNeutralItems() in d:\Build\gitextensions3_dev\TranslationApp\TranslationHelpers.cs:line 26
```

**Review by mstv — COMMENTED** [review]
~So, how do we get the `Atlassian.Jira.dll` into the right place?~
It exists in the folder `artifacts\Debug\bin\GitExtensions\net6.0-windows\Plugins\JiraCommitHintPlugin\'.
But it cannot be loaded. Is it targeting the wrong .NET?

**Comment by gerhardol** [comment]
> ~So, how do we get the `Atlassian.Jira.dll` into the right place?~ It exists in the folder `artifacts\Debug\bin\GitExtensions\net6.0-windows\Plugins\JiraCommitHintPlugin'. But it cannot be loaded. Is it targeting the wrong .NET?

Fixed by #10524 ?
If so, t5here is a cross dependency between the PRs.

**Comment by RussKie** [comment]
We don't currently build the plugin.  The only plausible explanation is
that you switched branches without cleaning the artifacts folder.

**Comment by mstv** [comment]
No, I got these errors after `git clean -xdf` as well.
But it works after merging #10524.

**Comment by mstv** [comment]
Since the correctness of `GitUI/Translation/English.Plugins.xlf` is checked by the AppVeyor build now, the PR #10524 must be merged first although it adds unnecessary stuff to `English.xlf` which will be cleaned up by this PR then.

**Comment by RussKie** [comment]
> Since the correctness of `GitUI/Translation/English.Plugins.xlf` is checked by the AppVeyor build now, the PR #10524 must be merged first although it adds unnecessary stuff to `English.xlf` which will be cleaned up by this PR then.

I disagree. https://github.com/gitextensions/gitextensions/pull/10524 is orthogonal to this change. And because it introduces irrelevant changes it must not go before this change.

File: GitUI/Translation/English.Plugins.xlf
Comment (RussKie) [reply GitUI/Translation/English.Plugins.xlf:1178]: Thank you for the update.

**Comment by RussKie** [comment]
> Any part of this for 4.0.x or do we leave those translations out of it? Or should we make a quick 4.1?

The removal of translations is ok, the updated strings _should_ be ok, I think. IIUIC, some translations are done by "machine translation", and hence it should be ok. If not, I will make sure we'll ship the languages we've already shipped in 4.0 even if the fall under the 95% threshold.

**Comment by RussKie** [comment]
> We don't currently build the plugin. The only plausible explanation is that you switched branches without cleaning the artifacts folder.

I stand corrected, we actually do build it and bundle it in... I was under the impression the plugin was removed from the build and the shippable artifacts.

**Pull request merged**
